design/seed_vm.md

```diff
@@ -0,0 +1,190 @@
+# SeedVM
+
+## Introduction
+
+SeedVM is a register based virtual machine which can interpret and execute the
+SeedLang byte code. Register based virtual machine avoids several push and pop
+instructions that stack based virtual machine needs to move values around the
+stack. These instructions are particularly expensive, because they involve copy
+of the values.
+
+The design intention of SeedVM is to make a stable, secure and fast virtual
+machine for the visualizable low-code programming environment that focuses on
+educational purposes.
+
+The SeedLang program is translated from SeedAST to byte code by the compiler.
+The design of the byte code is derived from the Lua 5.1. Please refer to [A
+No-Frills Introduction to Lua 5.1 VM
+Instructions](http://underpop.free.fr/l/lua/docs/a-no-frills-introduction-to-lua-5.1-vm-instructions.pdf).
+
+## SeedLang Value Types
+
+SeedLang is a dynamically typed programming language. Types are attached to
+values rather then variables. There are 9 types in SeedLang: Nil, Boolean,
+Number, String, List, Table, Function, Module and SeedObject:
+
+- Nil: Nil value, only one value nil
```

Comment:
I know it is a bit annoying but also worth doing this: add markdown code highlighting notation <code>\`...\`</code> to all identifiers, keywords, register names, and instruction names, so that the syntax highlighting will be way better than the default text style.

For example:

- `Nil`: Nil value, only one value `nil`.